src/test/java/teammates/test/cases/search/FeedbackResponseCommentSearchTest.java

```diff
@@ -76,9 +77,11 @@ public void allTests() {
         verifySearchResults(bundle, frc1I1Q2S1C1);
 
         ______TS("success: search for comments; confirms deleted comments are not included in results");
-
-        commentsDb.deleteDocument(commentsDb.getFeedbackResponseComment(frc1I3Q1S2C2.courseId, frc1I3Q1S2C2.createdAt,
-                frc1I3Q1S2C2.commentGiver));
+        FeedbackResponseCommentAttributes commentToDelete = commentsDb.getFeedbackResponseComment(frc1I3Q1S2C2.courseId,
+                frc1I3Q1S2C2.createdAt, frc1I3Q1S2C2.commentGiver);
+        commentsDb.deleteDocumentByCommentId(commentToDelete.feedbackResponseCommentId);
+        // search engine need time to persist changes
+        ThreadHelper.waitBriefly();
```

Comment:
This is quite brittle. Any better alternative?

Reply:
Use `persistenceRetryManager` to check instead.

The local search emulator also simulate the eventual consistency behaviour and there is no option to turn it off 😞